### PR TITLE
(PC-43060) feat(artist): use Wikipedia URL validity verification

### PR DESCRIPTION
**Pull request by clesausse2-pass**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-43060

## Flakiness

If I had to re-run tests in the CI due to flakiness, I add the incident [on Notion][2]

## Checklist

I have:

- [x] Made sure my feature is working on web.
- [x] Made sure my feature is working on mobile (depending on relevance : real or virtual devices)
- [x] Written **unit tests** native (and web when implementation is different) for my feature.
- [ ] Added a **screenshot** for UI tickets or deleted the screenshot section if no UI change
- [ ] If my PR is a bugfix, I add the link of the "résolution de problème sur le bug" [on Notion][1]
- [ ] I am aware of all the [best practices][3] and respected them.

[1]: https://www.notion.so/passcultureapp/R-solution-de-probl-mes-sur-les-bugs-5dd6df8f6a754e6887066cf613467d0a
[2]: https://www.notion.so/passcultureapp/cb45383351b44723a6f2d9e1481ad6bb?v=10fe47258701423985aa7d25bb04cfee&pvs=4
[3]: https://github.com/pass-culture/pass-culture-app-native/blob/master/doc/development/best-practices.md

### CODE REVIEW

File: src/features/artist/helpers/isWikipediaUrl.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+export function isWikipediaUrl(urlString?: string | null) {
```
Comment (bpeyrou-pass) [inline src/features/artist/helpers/isWikipediaUrl.ts]: On a déjà une fonction qui valide l'url, `isValidUrl` à spécialiser pour ton cas de figure, mais je préfèrerais qu'on réutilise ça histoire de pas dupliquer du code.

File: src/shared/isValidUrl/isValidUrl.ts [hunk 2]
```diff
@@ -12,3 +12,9 @@ export const isValidHttpsUrl = (identificationUrl: string, hostname?: string): b
 export const isValidUbbleUrl = (identificationUrl: string): boolean =>
   // needed for spy in tests cf. https://stackoverflow.com/questions/45111198/how-to-mock-functions-in-the-same-module-using-jest
   isValidUrl.isValidHttpsUrl(identificationUrl, 'id.ubble.ai')
+
+export const isValidWikipediaUrl = (urlString?: string | null): boolean => {
```
Comment (clesausse2-pass) [inline src/shared/isValidUrl/isValidUrl.ts]: Du coup, c'est moins complet que ma 1ère version mais je pense que c'est suffisant (si je me réfère a ce qui été fait pour Ubble)

File: src/shared/isValidUrl/isValidUrl.test.ts [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    expect(isValid).toBe(true)
+  })
+})
+
+describe('isValidWikipediaUrl', () => {
+  it('should return true for a standard french Wikipedia URL', () => {
+    expect(isValidUrl.isValidWikipediaUrl('https://fr.wikipedia.org/wiki/Victor_Hugo')).toBe(true)
+  })
+
+  describe('security and phishing prevention', () => {
```
Comment (bpeyrou-pass) [inline src/shared/isValidUrl/isValidUrl.test.ts]: Je préfère qu'on enlève cette description.

File: src/shared/isValidUrl/isValidUrl.test.ts [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    it('should return false for an empty string', () => {
+      expect(isValidUrl.isValidWikipediaUrl('')).toBe(false)
+    })
+
+    it('should return false for undefined or null', () => {
+      expect(isValidUrl.isValidWikipediaUrl(undefined)).toBe(false)
+      expect(isValidUrl.isValidWikipediaUrl(null)).toBe(false)
+    })
+
+    it('should return false for malformed or arbitrary string inputs', () => {
```
Comment (bpeyrou-pass) [inline src/shared/isValidUrl/isValidUrl.test.ts]: à priori c'est plutôt dans la fonction `isValidHttpsUrl` que ces tests devraient être

File: src/shared/isValidUrl/isValidUrl.test.ts [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+    })
+  })
+
+  describe('edge cases and invalid inputs', () => {
+    it('should return false for an empty string', () => {
+      expect(isValidUrl.isValidWikipediaUrl('')).toBe(false)
+    })
+
+    it('should return false for undefined or null', () => {
+      expect(isValidUrl.isValidWikipediaUrl(undefined)).toBe(false)
```
Comment (bpeyrou-pass) [inline src/shared/isValidUrl/isValidUrl.test.ts]: ne peut pas être `undefined` sur le typing il me semble

File: src/shared/isValidUrl/isValidUrl.test.ts [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+    it('should return false for malicious URI schemes (javascript, data, file)', () => {
+      expect(isValidUrl.isValidWikipediaUrl('javascript:alert("XSS")')).toBe(false)
+      expect(isValidUrl.isValidWikipediaUrl('data:text/html,<script>alert(1)</script>')).toBe(false)
+      expect(isValidUrl.isValidWikipediaUrl('file:///etc/passwd')).toBe(false)
+    })
+  })
+
+  describe('edge cases and invalid inputs', () => {
+    it('should return false for an empty string', () => {
+      expect(isValidUrl.isValidWikipediaUrl('')).toBe(false)
```
Comment (bpeyrou-pass) [inline src/shared/isValidUrl/isValidUrl.test.ts]: à décaler plutôt aussi vers la fonction `isValidHttpsUrl`

File: src/shared/isValidUrl/isValidUrl.test.ts [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+      expect(isValidUrl.isValidWikipediaUrl('https://attacker.com/fr.wikipedia.org')).toBe(false)
+      expect(isValidUrl.isValidWikipediaUrl('https://google.com/?search=wikipedia.org')).toBe(false)
+    })
+
+    it('should return false for typosquatted or non-.org domains', () => {
+      expect(isValidUrl.isValidWikipediaUrl('https://fake-wikipedia.org')).toBe(false)
+      expect(isValidUrl.isValidWikipediaUrl('https://wikipedia.com')).toBe(false)
+    })
+
+    it('should return false for malicious URI schemes (javascript, data, file)', () => {
```
Comment (bpeyrou-pass) [inline src/shared/isValidUrl/isValidUrl.test.ts]: déjà testé dans la `isValidHttpsUrl`

File: src/shared/isValidUrl/isValidUrl.test.ts (hunk 5 above)
Comment (clesausse2-pass) [reply src/shared/isValidUrl/isValidUrl.test.ts]:
```
export interface ArtistResponse {
  /**
   * @type {string | null}
   * @memberof ArtistResponse
   */
  descriptionSource?: string | null
```
Il peut côté artiste

File: src/shared/isValidUrl/isValidUrl.ts [hunk 8]
```diff
@@ -12,3 +12,9 @@ export const isValidHttpsUrl = (identificationUrl: string, hostname?: string): b
...
+  if (!urlString) return false
```
Comment (bpeyrou-pass) [inline src/shared/isValidUrl/isValidUrl.ts]: neatpicking mais tu peux retourner directement comme ça:
`export const isValidWikipediaUrl = (urlString?: string | null): boolean =>  !!(urlString && isValidUrl.isValidHttpsUrl(urlString, 'fr.wikipedia.org'))`

**Review by bpeyrou-pass — APPROVED** [review]
Je veux bien un dernier change plz
